src/Microsoft.Framework.CodeGeneration.Templating/Compilation/MetadataReferencesProvider.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Reflection;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.Framework.CodeGeneration.Templating.Compilation
+{
+    public class MetadataReferencesProvider
```

Comment:
Originally I created this class from the dnx451/dnxcore50-dependent parts of RoslynCompilationService. The idea was that I would have `#if DNX451 || DNXCORE50` that would surround the ILibraryManager and IApplicationEnvironment parts and do something different for net45. But when I did the net45 stuff I found that I could apply that to dnx451 & dnxcore50 as well and have a class that did not need the `#if'. If you like I could subsume this class back into RoslynCompilationService. However I thought it might be useful if someone needed to override just this service to provide their own references.